packages/beacon-node/src/sync/sync.ts

```diff
@@ -63,8 +63,23 @@ export class BeaconSync implements IBeaconSync {
       this.logger.debug("RangeSync disabled.");
     }
 
-    // TODO: It's okay to start this on initial sync?
-    this.chain.clock.on(ClockEvent.epoch, this.onClockEpoch);
+    // In case node is started with `rangeSync` disabled and `unknownBlockSync` is enabled.
+    // If the epoch boundary happens right away the `onClockEpoch` will check for the `syncDiff` and if
+    // it's more than 2 epoch will disable the disabling the `unknownBlockSync` as well.
+    // This will result into node hanging on the head slot and not syncing any blocks.
+    // This was the scenario in the test case `Unknown block sync` in `packages/cli/test/sim/multi_fork.test.ts`
+    // So we are adding a particular delay to ensure that the `unknownBlockSync` is enabled.
+    const syncStartSlot = this.chain.clock.currentSlot;
+    // Having one epoch time for the node to connect to peers and start a syncing process
+    const epochCheckFotSyncSlot = syncStartSlot + SLOTS_PER_EPOCH;
```

Comment:
I think one epoch would be enough time margin for a node to connect to peers and do the syncing before we start checking the sync progress. But this time could be changed based on node behavior.

Reply:
the comment and solution look great, I just don't feel right having this fix in prod code because `onClockEpoch` is designed to witness the genesis event as well.

Could you move this code to the above if/else code block for test code only? something like:

```typescript
if (!opts.disableRangeSync) {
      // prod code
      ...
     this.chain.clock.on(ClockEvent.epoch, this.onClockEpoch);
} else {
   // test code, this is needed for Unknown block sync sim test
   ...
   // TODO: fix sim test here
}
```